src/scripts/Viewer.vue

```diff
@@ -37,32 +37,65 @@ export default {
 	},
 	methods: {
 		thumbPath (item) {
-			let path = OC.joinPaths(
-				OC.linkToRemoteBase('dav/files'),
-				OC.getCurrentUser().uid,
-				item.path,
-				item.name
-			);
-
-			let params = $.param({
-				c: item.etag,
-				x: this.thumbDimensions,
-				y: this.thumbDimensions,
-				a: 1,
-				preview: 1
-			});
+
+			let path, params;
+
+			if (this.isPublic) {
+				path   = OC.getRootPath() + OC.filePath('files_sharing', 'ajax', 'publicpreview.php');
+				params = OC.buildQueryString({
+					file : item.path + '/' + item.name,
+					c: item.etag,
+					x: this.thumbDimensions,
+					y: this.thumbDimensions,
+					a: 1,
+					t : this.sharingToken
+				})
+			}
+
+			else {
+				path = OC.joinPaths(
+					OC.linkToRemoteBase('dav/files'),
+					OC.getCurrentUser().uid,
+					item.path,
```

Comment:
you need to encode these somehow.

I thought we had utility methods for that somewhere

Reply:
I am looking for those … didn't find any so far. I'll go ahead and look at what the video player did.

Reply:
Currently, path creation is super dirty. There is encoding in buildQueryString already, but joinPaths doesn't. Not sure, if I'm using all the wrong methods here.